Assets/Virtual Agents Framework/Runtime/Scripts/ScheduleBasedExecution/AgentTaskQueue.cs

```diff
@@ -82,6 +82,8 @@ private struct TaskEntry
         {
             public IAgentTask task;
             public int priority;
+            // May be null if the task is not part of a bundle
+            public TaskBundle taskBundle;
```

Comment:
Could we find a solution without an extra field here? Since the task bundles are AgentBaseTasks, we should ideally be able to handle them like normal tasks, so the bundle should be stored in the IAgentTask task field.

Reply:
Do we need to be able to check whether a task is a task bundle? For now i removed it as the attribute was not used anyway [(bffbff4)](https://github.com/rwth-acis/Virtual-Agents-Framework/pull/17/commits/bffbff4fe7a6d88ff9e8799d512dbe391b68b31f). It was meant to show the task bundle that the task belongs to.

Reply:
We can check whether the specific task object is derived from the TaskBundle class, e.g., using `myTask is TaskBundle` which returns true if myTask is an instance of TaskBundle or has a inherited type. For usability, it might indeed be relevant to provide some utility function for this but I would suggest to have this in some separate utility class. Something like `TaskUtilities.IsTaskBundle(myTask)`.